app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
+  def warnings
+    [].tap do |warnings|
+      warnings << Notice.new(problem_description_key, :warning) if problem?
+    end
+  end
+
+  def notices
+    statuses + warnings
+  end
+
+  # Filter to only status or warnings by passing `only: :status` or `only: :warning`
+  def badges_to_html(only: [:status, :warning])
```

Comment:
I don't think there'll be a significant performance hit. I'd actually be more worried about doing extra logic when we only want one (we do all the logic, but filter it), but these checks are all pretty snappy.